src/cpdb/helpers.py

```diff
@@ -22,31 +30,43 @@
 """a feedback from the group is to have a dictionary from the abbreviation for agency for something more explicity
 where would this list comes from? """
 
-
-def unpack_object(obj):
-    s = str(obj["Body"].read(), "utf8")
-    data = StringIO(s)
-    df = pd.read_csv(data, encoding="utf8")
-    return df
-
-
-def fetch_boto3_data(branch: str, table: str, previous=False):
-    if previous:
-        ver_key = f"db-cpdb/{branch}/2022-04-15/output/{table}.csv"
-    else:
-        ver_key = f"db-cpdb/{branch}/latest/output/{table}.csv"
-    client = boto3.client(
+def s3_resource():
+    return boto3.resource(
         "s3",
         aws_access_key_id=os.getenv("AWS_ACCESS_KEY_ID"),
         aws_secret_access_key=os.getenv("AWS_SECRET_ACCESS_KEY"),
         endpoint_url=os.getenv("AWS_S3_ENDPOINT"),
     )
-    obj = client.get_object(
-        Bucket="edm-private",
-        Key=ver_key,
+
+def get_csv(bucket, csv_filename):
+    obj = s3_resource().Object(bucket_name=bucket, key=csv_filename)
+    s = str(obj.get()["Body"].read(), "utf8")
+    data = StringIO(s)
+    return pd.read_csv(data, encoding="utf8")
+
+def zip_from_DO(zip_filename, bucket):
+    zip_obj = s3_resource().Object(bucket_name=bucket, key=zip_filename)
+    buffer = BytesIO(zip_obj.get()["Body"].read())
+
+    return ZipFile(buffer)
+
+def unzip_shapefile(zipfile, table):
+    try:
+        with zipfile as zf:
+            zf.extractall(path=f".library/{table}/")
```

Comment:
This is probably overkill for what this app is, but I think there's some edge cases to consider with putting this on the server.
For one, if two people view different branches at the same time here, the behavior is probably unpredictable (two extractalls to the same directory at the same time.) This could be fixed by adding the branch to the path here, but I think you're still probably creating race conditions that could be unpleasant. One thought I had was adding a timestamp as a folder and deleting it right after using it, like
```
# at the top
# from datetime import datetime
import shutil
# ...
            time = str(datetime.now().timestamp)
            zf.extractall(path=f".library/{time}/{table}/")
            df = gpd.read_file(f".library/{time}/{table}/{table}.shp")
            shutil.rmtree(path=f".library/{time}")
            return df
```
This way the files you're making are theoretically unique to you, and are deleted immediately

Reply:
haven't thought about this. Great catch.

Reply:
Not something I thought of but a really good point, I think we can make this into an issue and keep track of it probably not worth doing at this point (unless someone strongly objects) but a great future improvement

Reply:
It's not a super complicated so I did it in the most recent commit. Let me know if it works for you @mbh329 only caveat is you might have to comment out the withinNYC check because the geospatial file doesn't exist on any other branch

Reply:
@td928 Just to be clear, I should switch to a different branch and try to view the CPDB page but I have to comment out the `withinNYC` check in the `cpdb.py` file?